### ISSUE TITLE
how to set the test/val_dataset and test/val_iter

### ISSUE THREAD

**Issue by xiejiamingly** [SeitaroShinagawa/chainer-partial_convolution_image_inpainting#25]
When I try to train the model, in your train.py the code about setting test/val_dataset and test/val_iter was commented out，but it seem doesn't work when I remove the ’#‘ before the code，could you please tell me how to set the test/val_dataset and test/val_iter

**Comment by xiejiamingly** [comment]
![image](https://user-images.githubusercontent.com/13505282/50468208-87ac0500-09e1-11e9-8890-282124d0f546.png)
I set the test/val_dataset and test/val_iter like this.
when I run the train.py, after 100 iterations it stopped and show the following information:
![image](https://user-images.githubusercontent.com/13505282/50468259-c93cb000-09e1-11e9-9e70-edee8a6f07b8.png)
I don't know how to figure this out, could you please help me.

**Comment by SeitaroShinagawa** [comment]
Thank you for your report. This bug happens if your dataloader fails to load samples.
For using val/test set, could you try to use `place2_test` in `dataset/place.py`?  
```python
# in train.py
    ...
    val_dataset = getattr(datasets, "place2_test")(paths.val_place2, mask_path="mask/256", flip=0, 
    resize_to=args.resize_to, crop_to=args.crop_to)
    val_iter = chainer.iterators.SerialIterator(val_dataset, 8)

    # Set up a trainer
    updater = Updater(
        models=(vgg, model),
        iterator={
            'main': train_iter,
            'test': val_iter
            },
```

I also found a bug in `place2_test`  
```python
self.trainAkey = glob.glob(dataset_path + "*.jpg") # wrong
self.trainAkey = glob.glob(dataset_path + "/*.jpg") # correct
```  
I have already fix them in the latest version. You can see the all diffs in 303cc00ff03503a8eab22c933f24cd3332437d53

**Comment by xiejiamingly** [comment]
Thanks!
